smithy-utils/src/main/java/software/amazon/smithy/utils/CodeWriter.java

```diff
@@ -944,6 +945,43 @@ public final CodeWriter write(Object content, Object... args) {
         return this;
     }
 
+    /**
+     * Writes text to the CodeWriter without appending a newline or prefixing indentation.
+     *
+     * <p>If newlines are present in the given string, each of those lines will receive proper indentation.
+     *
+     * @param content Content to write.
+     * @param args String arguments to use for formatting.
+     * @return Returns the CodeWriter.
+     */
+    public final CodeWriter writeInline(Object content, Object... args) {
+        String value = formatter.format(content, currentState.indentText, this, args);
+        ArrayList<String> lines = new ArrayList<>(Arrays.asList(value.split(newlineRegexQuoted, -1)));
```

Comment:
(Probably) nit: Does this need to materialize a `List`? It's probably not a big deal, but if this is called a lot, then it's a lot of unnecessary garbage to collect. Or could this even just change to replace the newline with the newline current indentation + string? We're already loading the whole thing into memory in a string.